packages/react/src/components/Masthead/Masthead.js

```diff
@@ -259,6 +259,7 @@ const Masthead = ({
               {(mastheadL1Data || navigation) && (
                 <HeaderMenuButton
                   aria-label={isSideNavExpanded ? 'Close menu' : 'Open menu'}
+                  id={`${stablePrefix}--masthead__hamburger`}
```

Comment:
@kennylam Wondering this is an attempt to solve a similar problem to #3660?

Reply:
@asudoh Yes, was just about to leave a comment in your PR. The masthead `autoid` is now variable, having `default`, `eco`, etc, types. I think it will be more reliable to target this way.